virtual_machine.go

```diff
@@ -643,3 +643,20 @@ func (v *VirtualMachine) ConvertToTemplate(ctx context.Context) (task *Task, err
 	}
 	return NewTask(upid, v.client), nil
 }
+
+func (v *VirtualMachine) UnmountCloudInitISO(ctx context.Context, device string) error {
+	if v.HasTag(MakeTag(TagCloudInit)) {
+		if ok, err := v.deleteCloudInitISO(ctx); err != nil || !ok {
```

Comment:
this returns *Task and I dont think =the `!ok` check is doing what you think here. If you actually want to make sure the ISO deletes then WaitFor the task, if you don't care if the ISO deletes you can ignore the Task with `_` and check `err != nil` and you'll know the proxmox api received the request and will be trying to delete the ISO in the background.

Reply:
yes